includes/acl/decompression/stream/read_stream_data.h

```diff
@@ -520,44 +518,69 @@ namespace acl
 	}
 
 	template<class SettingsAdapterType, class DecompressionContext>
-	inline void decompress_vector(const SettingsAdapterType& settings, const ClipHeader& header, DecompressionContext& context, Vector4_32* out_vectors) { decompress_vectors<1>(settings, header, context, out_vectors); }
+	inline void decompress_vector(const SettingsAdapterType& settings, const ClipHeader& header, DecompressionContext& context, Vector4_32* out_vectors, TimeSeriesType8& out_time_series_type) { decompress_vectors<1>(settings, header, context, out_vectors, out_time_series_type); }
 
 	template<class SettingsAdapterType, class DecompressionContext>
-	inline void decompress_vectors_in_two_key_frames(const SettingsAdapterType& settings, const ClipHeader& header, DecompressionContext& context, Vector4_32* out_vectors) { decompress_vectors<2>(settings, header, context, out_vectors); }
+	inline void decompress_vectors_in_two_key_frames(const SettingsAdapterType& settings, const ClipHeader& header, DecompressionContext& context, Vector4_32* out_vectors, TimeSeriesType8& out_time_series_type) { decompress_vectors<2>(settings, header, context, out_vectors, out_time_series_type); }
 
 	template<class SettingsAdapterType, class DecompressionContext>
-	inline void decompress_vectors_in_four_key_frames(const SettingsAdapterType& settings, const ClipHeader& header, DecompressionContext& context, Vector4_32* out_vectors) { decompress_vectors<4>(settings, header, context, out_vectors); }
+	inline void decompress_vectors_in_four_key_frames(const SettingsAdapterType& settings, const ClipHeader& header, DecompressionContext& context, Vector4_32* out_vectors, TimeSeriesType8& out_time_series_type) { decompress_vectors<4>(settings, header, context, out_vectors, out_time_series_type); }
 
 	template <class SettingsType, class DecompressionContext>
 	inline Quat_32 decompress_and_interpolate_rotation(const SettingsType& settings, const ClipHeader& header, DecompressionContext& context)
 	{
 		Quat_32 rotations[2];
+		TimeSeriesType8 time_series_type;
 
 		ACL_ENSURE(array_length(context.key_frame_byte_offsets) == array_length(rotations), "Interpolation requires exactly two keyframes.");
 
-		decompress_rotations_in_two_key_frames(settings, header, context, rotations);
+		decompress_rotations_in_two_key_frames(settings, header, context, rotations, time_series_type);
```

Comment:
I'm not a fan of this. It's likely to branch for no reason. It might make more sense to pass the interpolation function as a template argument (or adapter struct) and to call it when necessary (e.g. varying). This will also avoid returning 2 rotations although it's hard to say how the generated code will look like once/if inlined

Reply:
I was hoping to separate the interpolation from the decompression of the rotations themselves, since a lerp might not be what you want. For example, for cubics I might need to decompress four consecutive rotations to first initializing the context, but I don't want to interpolate at that time. Similarly if the user seeks forward a little, a single rotation might need to be decompressed [and shifted into the "working set"] but the interpolation will involve four rotations.